pkgs/applications/radio/anytone-emu/default.nix

```diff
@@ -0,0 +1,45 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, udev
+, stdenv
+, darwin
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "anytone-emu";
+  version = "unstable-2023-06-15";
+
+  src = fetchFromGitHub {
+    owner = "hmatuschek";
+    repo = "anytone-emu";
+    rev = "c6a63b1c9638b48ed0969f90a5e11e2a5fe59458";
+    hash = "sha256-Y+7DkenYiwnfVWtMwmtX64sUN7bBVoReEmZQfEjHn8o=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+  };
+
+  postPatch = ''
+    ln -s ${./Cargo.lock} Cargo.lock
+  '';
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    udev
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.CoreFoundation
```

Comment:
```suggestion
```
I think this application only works on Linux.
It calls a Linux-specific function internally.
https://docs.rs/nix/latest/nix/pty/fn.ptsname_r.html